app/models/host/managed_extensions.rb

```diff
@@ -27,7 +27,7 @@ def queue_reboot
   end
 
   def setReboot
-    old.becomes(Host::Discovered).reboot(facts["discovery_bootip"] || facts["ipaddress"])
+    old.becomes(Host::Discovered).reboot(old.ip, ip)
```

Comment:
Root cause was that `old` and `self` are only available for host (this class is included into host). Not later.